File: leakybucket.py
import time

n=3
i=200
o=50
bs=300
s=0

while(n!=0):
	print("incoming packets :",i)
	if(i<=(bs-s)):
		print("no packets lost ")
		s=s+i
		print("Bucketsize :"+str(bs)+"Store wihtout packet loss:"+str(s))
	else:
		print("packets lost"+str(i-(bs-s)))
		s=bs
		print("Bucketsize :"+str(bs)+"Store with packet loss:"+str(s))
	s=s-o
	print("Bucketsize :"+str(bs)+"Store after outgoing:"+str(s))

	n=n-1
	time.sleep(5)



		 
		

File: leakybucket.cc
#include<iostream>
using namespace std;

int main()
{
int n=4,store=0,size=250,out_rate=50;
int in_rate[4]={100,200,100,150};

cout<<n<<store<<size<<out_rate;

for(int i=0;i<4;i++)
{
cout<<in_rate[i];
}

while(n!=0)
{
for(int i=0;i<4;i++)
{
if(in_rate[i]<(size-store))
{cout<<"packets not lost";
store+=in_rate[i];
cout<<store<<" "<<size;}
else
{
cout<<"packets lost";
store=size;
cout<<store<<" "<<size;
}

if(store>0)
{
store-=out_rate;
cout<<store<<" "<<size<<"After outgoing";
}
else
cout<<"EMPTY ";

n--;
}
}

while(store>0)
{

store-=out_rate;

cout<<store<<"after all the inputs are taken";
}


}


